<launch>
  <!--
  <arg name="bag_dir" default="/home/chalmers/bagfiles/gyro_correction" />
  <arg name="file" default="2018-03-13-14-29-50_0.bag" /> 

  <node pkg="tf" type="tf_remap" name="tf_remapper" output="screen">
    <rosparam param="mappings">
      [{old: base_link, new: /base_link_bag},{old: base_stabilized, new: /base_stabilized_bag}]
    </rosparam>
  </node>
here minusminus clock
  <node pkg="rosbag" type="play" name="player" args="  clock $(arg bag_dir)/$(arg file)">
    <remap from="tf" to="tf_old" />
  </node> 
-->
  <param name="use_sim_time" value="true"/>

  <!--<node pkg="div_datalogger_postprocess" type="correct_gyro_drift.py"></node>-->
  
  <node pkg="nodelet" type="nodelet" name="imu_manager" args="manager" output="screen" />
  <!--<node pkg="imu_filter_madgwick" type="imu_filter_node" name="imu_filter_node" output="screen">-->
  <node pkg="nodelet" type="nodelet" name="ImuFilterNodelet" args="load imu_filter_madgwick/ImuFilterNodelet imu_manager" output="screen">
    <remap from="/imu/data_raw" to="/imu/data_raw_corrected"/>
    <remap from="/imu/data" to="/imu/data_corrected"/>
    <param name="world_frame" value="nwu"/>
    <param name="fixed_frame" value="base_stabilized_corrected"/>
    <param name="use_mag" value="true"/>
    <param name="use_magnetic_field_msg" value="true"/>
  </node>

  <node pkg="rosbag" type="play" name="player" args="/home/chalmers/bagfiles/gyro_correction/2018-03-13-15-13-59_0_test.bag --delay 4 -r 2 --clock"></node>

  <node pkg="rosbag" type="record" name="recorder" args="-a -O /home/chalmers/bagfiles/gyro_correction/2018-03-13-15-13-59_0_CORRECT.bag"></node>



</launch>